tests/Unit/SDK/Resource/Detectors/EnvironmentTest.php

```diff
@@ -35,19 +41,38 @@ public function test_environment_get_resource_with_resource_attributes(): void
     {
         $this->setEnvironmentVariable('OTEL_RESOURCE_ATTRIBUTES', 'key_foo=value_foo,key_bar=value_bar');
 
-        $resouceDetector = new Detectors\Environment();
-        $resource = $resouceDetector->getResource();
+        $resource = $this->detector->getResource();
 
         $this->assertSame('value_foo', $resource->getAttributes()->get('key_foo'));
         $this->assertSame('value_bar', $resource->getAttributes()->get('key_bar'));
     }
 
+    /**
+     * @dataProvider encodedResourceValueProvider
+     */
+    public function test_environment_get_resource_with_encoded_value(string $value, string $expected): void
+    {
+        $key = '_key';
```

Comment:
just curious, any reason of using underscore prefix?

Reply:
no good reason. I've removed it to avoid any confusion.